keps/sig-node/20191025-image-pull-progress.md

```diff
@@ -0,0 +1,385 @@
+
+## Release Signoff Checklist
+
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before [Enhancement
+Freeze](https://github.com/kubernetes/sig-release/tree/master/releases) of the
+targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP
+      (this should be a link to the KEP location in kubernetes/enhancements, not the
+      initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG
+      Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for
+      publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to
+      mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once
+it is marked `implementable` should be approved by each of the KEP approvers. If
+any of those approvers is no longer appropriate than changes to that list should
+be approved by the remaining approvers and/or the owning SIG (or SIG-arch for
+cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every
+time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Target of this enhancement is to expose the progress of a container image pull
+to the end user via the API and the CLI. When a Kubernetes workload gets
+created, then the user has now the possibility to obtain additional information,
+like the overall amount of bytes already downloaded, the needed bytes to be
+downloaded and the estimated pull time based on the currently available
+bandwidth. These information are directly available via tools like `kubectl`,
+because each workload exposes this state directly via an API.
+
+## Motivation
+
+During Kubernetes workload creation, the time needed to pull the container image
+from the remote location varies depending on multiple factors. In low-bandwidth
+scenarios the image pull step might consume a significant amount time
```

Comment:
I'd love to learn more about the scenarios in which this feature would be most useful and how often they occur. When you get a chance, would you mind commenting on the following:

- Is this request one we've heard from users of k8s?
- How often do operators actively monitor a container image being pulled from a remote location?
- What would happen if we didn't undertake work on this feature? I.e. what are the risks to k8s and its applications?

Reply:
Actually, I answered my own questions reading https://github.com/kubernetes/kubernetes/issues/19077 :)

Sounds like something the community is interested in!

If there's additional thought you have on my questions, feel free to share.